controller/src/lib.rs

```diff
@@ -21,12 +21,58 @@ pub mod run;
 mod scheduler;
 
 pub async fn run_scheduler(nats: TypedNats, state: StateHandle) -> NeverResult {
-    let scheduler = Scheduler::new(state);
+    let scheduler = Scheduler::new(state.clone());
     let mut schedule_request_sub = nats.subscribe(ScheduleRequest::subscribe_subject()).await?;
     tracing::info!("Subscribed to spawn requests.");
 
     while let Some(schedule_request) = schedule_request_sub.next().await {
         tracing::info!(spawn_request=?schedule_request.value, "Got spawn request");
+
+        if let Some(lock) = &schedule_request.value.lock {
+            tracing::info!(lock=%lock, "Request includes lock.");
+            let locked = {
+                let state = state.state();
+                let cluster = state
+                    .cluster(&schedule_request.value.cluster)
+                    .ok_or_else(|| anyhow!("Cluster does not exist."))?;
```

Comment:
If the schedule request includes a cluster that does not exist, will this end up returning and stopping the `run_scheduler` loop?

Reply:
Yes. It's not the end of the world now that the scheduler runs inside a `Supervisor`, but it should really just log it and continue.